imports/ui/components/ShoppingListButton.jsx

```diff
@@ -1,18 +1,149 @@
+import "./ShoppingListButton.css";
 
 class ShoppingListButton extends Component {
   constructor(props) {
     super(props);
+    this.state = { left: false, checked: this.props.checkList.checkList };
+  }
+
+  componentDidMount() {
+    this.props.getShoppingListItems();
   }
 
+  toggleDrawer = (side, open) => event => {
+    if (
+      event.type === "keydown" &&
+      (event.key === "Tab" || event.key === "Shift")
+    ) {
+      return;
+    }
+    this.props.getCheckListItems();
+    this.setState({checked: this.props.checkList.checkList});
+    this.setState({ ...this.state, [side]: open });
+  };
+  handleToggle = value => () => {
+    const currentIndex = this.props.checkList.checkList.indexOf(value);
+    const newChecked = [...this.props.checkList.checkList];
+    if (currentIndex === -1) {
+      newChecked.push(value);
+    } else {
+      newChecked.splice(currentIndex, 1);
+    }
+    this.props.addToCheckedList(newChecked);
+  };
+
+  sideList = (side, items) => (
+    <div
+      role="presentation"
+      onKeyDown={this.toggleDrawer(side, false)}
+      className="list-sector"
+    >
+      <Typography variant="h6" id="shopping-title">
+        Your personalized shopping list
+      </Typography>
+      <Divider />
+      <List>
+        {this.props.items.shoppingList.length === 0 ? (
+          <IconButton>Start adding items to your shopping list!</IconButton>
+        ) : (
+          ""
+        )}
+        {items.map((text, index) => {
+          const labelId = `checkbox-list-secondary-label-${index}`;
+          return (
+            <ListItem key={index}>
+              <ListItemAvatar>
+                <Avatar alt="Remy Sharp" src={text.imageSrc} />
+              </ListItemAvatar>
+              <ListItemText
+                primary={text.name}
+                secondary={
+                  <React.Fragment>
+                    <Typography
+                      component="span"
+                      variant="body2"
+                      color="textPrimary"
+                    >
+                      {"$ " + text.price + " " + text.unit}
+                    </Typography>
+                    {" " + text.location.address}
+                  </React.Fragment>
+                }
+              />
+              <ListItemSecondaryAction>
+                <Checkbox
+                  edge="end"
+                  onChange={this.handleToggle(index)}
+                  checked={this.props.checkList.checkList.indexOf(index) !== -1}
+                  inputProps={{ "aria-labelledby": labelId }}
+                />
+              </ListItemSecondaryAction>
+            </ListItem>
+          );
+        })}
+      </List>
+    </div>
+  );
+
   render() {
+    const items = this.props.items.shoppingList;
     return (
-      <span>{this.props.currentUser && 
-      <Button>ShoppingList</Button>}</span>
+      <span>
+        {this.props.currentUser && (
+          <span>
+                    <Button
```

Comment:
Did you run prettier?

Reply:
Yeah but it didn't fix. I'll move it back a little bit and then run prettier again.